File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/service/MyMediaProjectionService.java
package com.stringee.apptoappcallsample.service;

import android.app.Notification;
import android.app.Service;
import android.content.Intent;
import android.content.pm.ServiceInfo;
import android.os.Build;
import android.os.IBinder;

import androidx.annotation.Nullable;
import androidx.core.app.ServiceCompat;

import com.stringee.apptoappcallsample.common.Constant;
import com.stringee.apptoappcallsample.common.NotificationUtils;
import com.stringee.apptoappcallsample.common.Utils;
import com.stringee.apptoappcallsample.manager.CallManager;

public class MyMediaProjectionService extends Service {
    private CallManager callManager;

    @Override
    public void onCreate() {
        super.onCreate();
        callManager = CallManager.getInstance(this);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent != null) {
            String action = intent.getAction();
            if (!Utils.isStringEmpty(action)) {
                if (action.equals(Constant.ACTION_START_FOREGROUND_SERVICE)) {
                    Notification notification = NotificationUtils.getInstance(this).createMediaNotification();
                    int type = 0;
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                        type = ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PROJECTION;
                    }
                    try {
                        ServiceCompat.startForeground(this, Constant.MEDIA_SERVICE_ID, notification, type);
                    } catch (Exception e) {
                        Utils.reportException(MyMediaProjectionService.class, e);
                    }
                    callManager.startCapture(this, intent);
                }
            }
        }
        return START_NOT_STICKY;
    }

    public void stopService() {
        stopForeground(STOP_FOREGROUND_REMOVE);
        stopSelf();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/service/MyMediaProjectionService.kt
package com.stringee.kotlin_onetoonecallsample.service

import android.app.Service
import android.content.Intent
import android.content.pm.ServiceInfo
import android.os.Build
import android.os.IBinder
import androidx.core.app.ServiceCompat
import com.stringee.kotlin_onetoonecallsample.common.Constant
import com.stringee.kotlin_onetoonecallsample.common.NotificationUtils
import com.stringee.kotlin_onetoonecallsample.common.PrefUtils
import com.stringee.kotlin_onetoonecallsample.common.Utils
import com.stringee.kotlin_onetoonecallsample.manager.CallManager

class MyMediaProjectionService : Service() {
    private lateinit var callManager: CallManager
    override fun onCreate() {
        super.onCreate()
        callManager = CallManager.getInstance(this)
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        val action = intent.action
        if (!Utils.isStringEmpty(action)) {
            if (action == Constant.ACTION_START_FOREGROUND_SERVICE) {
                val notification =
                    NotificationUtils.getInstance(this).createMediaNotification()
                var type = 0
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                    type = ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PROJECTION
                }
                try {
                    ServiceCompat.startForeground(
                        this,
                        Constant.MEDIA_SERVICE_ID,
                        notification,
                        type
                    )
                } catch (e: Exception) {
                    Utils.reportException(CallManager::class.java, e)
                }
                callManager.startCapture(this)
            }
        }
        return START_NOT_STICKY
    }

    fun stopService() {
        stopForeground(STOP_FOREGROUND_REMOVE)
        stopSelf()
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }
}